test/src/lib/lockfile.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { expect } from 'chai';
+import { dirname, basename } from 'path';
+import { stub, SinonStub } from 'sinon';
+import { promises as fs } from 'fs';
+import mock = require('mock-fs');
+
+import * as lockfile from '../../../src/lib/lockfile';
+import * as fsUtils from '../../../src/lib/fs-utils';
+
+describe('lib/lockfile', () => {
+	const lockPath = '/tmp/balena-supervisor/services/1234567/updates.lock';
+	// mock-fs expects an octal file mode, however, Node's fs.stat.mode returns a bit field:
+	//   - 16877 (Node) == octal 0755 (drwxr-xr-x)
+	//   - 17407 (Node) == octal 1777 (drwxrwxrwt)
+	const DEFAULT_PERMISSIONS = {
+		unix: parseInt('0755', 8),
+		node: 16877,
+	};
+	const STICKY_WRITE_PERMISSIONS = {
+		unix: parseInt('1777', 8),
+		node: 17407,
+	};
+	const NOBODY_UID = 65534;
+
+	const mockDir = (
+		mode: number = DEFAULT_PERMISSIONS.unix,
+		opts: { createLock: boolean } = { createLock: false },
+	) => {
+		const items: any = {};
+		if (opts.createLock) {
+			items[basename(lockPath)] = mock.file({ uid: NOBODY_UID });
+		}
+
+		mock({
+			[dirname(lockPath)]: mock.directory({
+				mode,
+				items,
+			}),
+		});
+	};
+
+	const checkDirPermissions = async (path: string, permissions: number) => {
+		expect((await fs.stat(dirname(path))).mode).to.equal(permissions);
+	};
+
+	const checkLockDirFiles = async (
+		path: string,
+		opts: { shouldExist: boolean } = { shouldExist: true },
+	) => {
+		const files = await fs.readdir(dirname(path));
+		if (opts.shouldExist) {
+			expect(files).to.include(basename(path));
+		} else {
+			expect(files).to.have.length(0);
+		}
+	};
+
+	let execStub: SinonStub;
+
+	beforeEach(() => {
+		// @ts-ignore
+		execStub = stub(fsUtils, 'exec').callsFake(async (command, opts) => {
+			// Extract target path
+			const targetPath = command.replace('lockfile -r 0 ', '');
```

Comment:
Although this works, I don't like that it is depending on an internal implementation detail of the lockfile module. If we ever want to add another argument to the `lockfile` call then this will fail, which goes against the idea of tests being resistant to refactoring. 

It's a bit extra, but maybe something like this?

```suggestion
                        // Sanity check for the command call
                        expect(command.trim().startsWith('lockfile')).to.be.true
                        
                        // Remove any command options to leave just the command and the target file
			const [,targetPath] = command.replace(/-v|-nnn|-r\s+\d+|-l\s+\d+|-s\s+\d+|-!|-ml|-mu/g,'').split(/\s+/);
```

Reply:
No arguments against this.